webapp/tests/karma/ts/modules/reports/reports.component.spec.ts

```diff
@@ -263,4 +278,156 @@ describe('Reports Component', () => {
     });
   });
 
+  describe('Reports breadcrumbs', () => {
+    it('it should retrieve the hierarchy level of the connected user', () => {
+      expect(component.currentLevel).to.equal('parent');
+    });
+
+    it('should not change the reports lineage if user is online only', async () => {
+      const reports = [
+        {
+          _id: '88b0dfff-4a82-4202-abea-d0cabe5aa9bd',
+          lineage: [ 'St Elmos Concession', 'Chattanooga Village', 'CHW Bettys Area' ],
+        },
+        {
+          _id: 'a86f238a-ad81-4780-9552-c7248864d1b2', lineage:  [ 'Chattanooga Village', 'CHW Bettys Area'],
+        },
+        {
+          _id: 'd2da792d-e7f1-48b3-8e53-61d331d7e899', lineage: [ 'Chattanooga Village' ],
+        },
+        {
+          _id: 'ee21ea15-1ebb-4d6d-95ea-7073ba357229', lineage: [ 'CHW Bettys Area'],
+        },
+        {
+          _id: 'ee21ea15-1ebb-4d6d-95ea-7073ba357229', lineage: [],
+        },
+        {
+          _id: 'ee21ea15-1ebb-4d6d-95ea-7073ba965525',
+        },
+      ];
+      const expectedReports = [
+        {
+          _id: '88b0dfff-4a82-4202-abea-d0cabe5aa9bd', 
+          lineage: [ 'St Elmos Concession', 'Chattanooga Village', 'CHW Bettys Area' ],
+          heading: 'report.subject.unknown',
+          icon: undefined,
+          summary: undefined,
+          unread: true,
+        },
+        {
+          _id: 'a86f238a-ad81-4780-9552-c7248864d1b2', 
+          lineage:  [ 'Chattanooga Village', 'CHW Bettys Area' ],
+          heading: 'report.subject.unknown',
+          icon: undefined,
+          summary: undefined,
+          unread: true,
+        },
+        {
+          _id: 'd2da792d-e7f1-48b3-8e53-61d331d7e899',
+          lineage: [ 'Chattanooga Village' ],
+          heading: 'report.subject.unknown',
+          icon: undefined,
+          summary: undefined,
+          unread: true,
+        },
+        {
+          _id: 'ee21ea15-1ebb-4d6d-95ea-7073ba357229', 
+          lineage: [ 'CHW Bettys Area' ],
+          heading: 'report.subject.unknown',
+          icon: undefined,
+          summary: undefined,
+          unread: true,
+        },
+        {
+          _id: 'ee21ea15-1ebb-4d6d-95ea-7073ba357229',
+          lineage: [],
+          heading: 'report.subject.unknown',
+          icon: undefined,
+          summary: undefined,
+          unread: true,
+        },
+        {
+          _id: 'ee21ea15-1ebb-4d6d-95ea-7073ba965525',
+          lineage: undefined,
+          heading: 'report.subject.unknown',
+          icon: undefined,
+          summary: undefined,
+          unread: true,
+
+        },
+      ];
+      userContactService.get.resolves(userContactDoc);
```

Comment:
Maybe I am missing something here, but don't we need to use a user whose parent is actually contained in the lineage?  (So that we can verify the lineage was not actually edited because the user is online?)

Reply:
As I was explaining [here](https://github.com/medic/cht-core/pull/7732#discussion_r964392778). 
When the user is online only, component.currentLevel is not even set in the component, so calling component.currentLevel = await component.getCurrentLineageLevel() is not necessary.

Would you be confortable with something like this ? :
    it('should not change the reports lineage if user is online only', async () => {
      const expectedReports = [
        {
          _id: '88b0dfff-4a82-4202-abea-d0cabe5aa9bd',
          lineage: [ 'St Elmos Concession', 'Chattanooga Village', 'CHW Bettys Area' ],
          heading: 'report.subject.unknown',
          icon: undefined,
          summary: undefined,
          unread: true,
        },
        {
          _id: 'a86f238a-ad81-4780-9552-c7248864d1b2',
          lineage:  [ 'Chattanooga Village', 'CHW Bettys Area' ],
          heading: 'report.subject.unknown',
          icon: undefined,
          summary: undefined,
          unread: true,
        },
        {
          _id: 'd2da792d-e7f1-48b3-8e53-61d331d7e899',
          lineage: [ 'Chattanooga Village' ],
          heading: 'report.subject.unknown',
          icon: undefined,
          summary: undefined,
          unread: true,
        },
        {
          _id: 'ee21ea15-1ebb-4d6d-95ea-7073ba357229',
          lineage: [ 'CHW Bettys Area' ],
          heading: 'report.subject.unknown',
          icon: undefined,
          summary: undefined,
          unread: true,
        },
        {
          _id: 'ee21ea15-1ebb-4d6d-95ea-7073ba357229',
          lineage: [],
          heading: 'report.subject.unknown',
          icon: undefined,
          summary: undefined,
          unread: true,
        },
        {
          _id: 'ee21ea15-1ebb-4d6d-95ea-7073ba965525',
          lineage: undefined,
          heading: 'report.subject.unknown',
          icon: undefined,
          summary: undefined,
          unread: true,

        },
      ];
      // userContactService.get.resolves(userContactDoc);
      // sessionService.isOnlineOnly.resolves(true);

      component.currentLevel = undefined;    //await component.getCurrentLineageLevel();

      const updatedReports = component.prepareReports(reports);

      // expect(component.currentLevel).to.equal('parent');

      expect(updatedReports).to.deep.equal(expectedReports);
    });